### ISSUE TITLE
addMarker in Google Maps stopped working

### ISSUE THREAD

**Issue by ColinFrick** [driftyco/ionic-native#658]
I'm unable to add any marker to the map. If I call addMarker I receive following error in LogCat:

```
10-07 18:06:49.487: W/System.err(3799): java.lang.reflect.InvocationTargetException
10-07 18:06:49.487: W/System.err(3799): 	at java.lang.reflect.Method.invoke(Native Method)
10-07 18:06:49.487: W/System.err(3799): 	at plugin.google.maps.MyPlugin.execute(MyPlugin.java:49)
10-07 18:06:49.487: W/System.err(3799): 	at plugin.google.maps.GoogleMaps$2.run(GoogleMaps.java:259)
10-07 18:06:49.487: W/System.err(3799): 	at android.os.Handler.handleCallback(Handler.java:739)
10-07 18:06:49.487: W/System.err(3799): 	at android.os.Handler.dispatchMessage(Handler.java:95)
10-07 18:06:49.487: W/System.err(3799): 	at android.os.Looper.loop(Looper.java:148)
10-07 18:06:49.487: W/System.err(3799): 	at android.app.ActivityThread.main(ActivityThread.java:5417)
10-07 18:06:49.487: W/System.err(3799): 	at java.lang.reflect.Method.invoke(Native Method)
10-07 18:06:49.487: W/System.err(3799): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
10-07 18:06:49.487: W/System.err(3799): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
10-07 18:06:49.487: W/System.err(3799): Caused by: org.json.JSONException: No value for lat
10-07 18:06:49.487: W/System.err(3799): 	at org.json.JSONObject.get(JSONObject.java:389)
10-07 18:06:49.487: W/System.err(3799): 	at org.json.JSONObject.getDouble(JSONObject.java:444)
10-07 18:06:49.487: W/System.err(3799): 	at plugin.google.maps.PluginMarker.createMarker(PluginMarker.java:57)
10-07 18:06:49.487: W/System.err(3799): 	... 10 more
```

The properties lat and lng are not added directly to the position object, but are wrapped in _objectInstance:
`10-07 18:06:49.486: I/System.out(3799): {"_objectInstance":{"lat":43,"lng":-89}}`

Test source code:
```
        let ionic: GoogleMapsLatLng = new GoogleMapsLatLng(43, -89);
        let markerOptions: GoogleMapsMarkerOptions = {
            position: ionic,
            title: 'Ionic'
        };

        this.map.addMarker(markerOptions)
            .then((marker: GoogleMapsMarker) => {
                console.log("added Marker");
                marker.showInfoWindow();
            })
            .catch(() => console.log("encountered an error while adding a marker"));
```

Tested in the Android Emulator: 
- Android 6.0 (Google APIs)
- Google Maps 1.3.9 & Latest

**Comment by ZiFFeL1992** [comment]
Have you tried with branch master?

Also offtopic (sorry): How did you get those debug logs?

**Comment by ColinFrick** [comment]
The master branch of ionic-native?
As I'm using the latest release 2.1.7 and that was released 26 hours ago, I doubt that would change something but I'll try.

If you mean the google maps master, yes I tried both npm and directly from git.

Off topic: If you use the Android Studio you'll get the Android Device Monitor, which is able to connect to the emulator and show the log files of the ionic app. Really handy.

**Comment by ZiFFeL1992** [comment]
I was refering to branch master of Google Maps plugin.

How many markers are you adding? Does the code fails when you add the first marker?

**Comment by ZiFFeL1992** [comment]
Try this:

```
 this.map.addMarker({
   'position': ionic,
   'title': 'ionic',
   markerClick: (marker) => {
     marker.showInfoWindow();
   },
   infoClick: (marker) => {
     console.log('info clicked');
   }
});
```

**Comment by ColinFrick** [comment]
Actually only one. All properties defined as GoogleMapsLatLng don't work for me.
Even setting up GoogleMap with the camera property doesn't work.

I created a new ionic2 project, and followed http://www.joshmorony.com/integrating-native-google-maps-into-an-ionic-2-application/ to reproduce it.

This throws an error:
```
let location = new GoogleMapsLatLng(-34.9290,138.6010);

this.map = new GoogleMap('map', {
  'backgroundColor': 'white',
  'controls': {
    'compass': true,
    'myLocationButton': true,
    'indoorPicker': true,
    'zoom': true
  },
  'gestures': {
    'scroll': true,
    'tilt': true,
    'rotate': true,
    'zoom': true
  },
  'camera': {
    'latLng': location,
    'tilt': 30,
    'zoom': 15,
    'bearing': 50
  }
});
```

Removing the 'camera' property fixes the problem.

**Comment by ColinFrick** [comment]
Reverting ionic-native to the default version 2.0.3 fixes the problem (https://github.com/driftyco/ionic2-app-base/blob/master/package.json)

**Comment by ramonornela** [comment]
Try using branch master of the plugin compatible with wkwebview and cordova 5 in iOS, maybe the code was adjusted to this.

Thanks

**Comment by ihadeed** [comment]
This could be caused by this edit here: https://github.com/driftyco/ionic-native/commit/72a694a5e1a61d8743e742ebe7431bafd3f0ed4d#diff-7380fd903dab74ddcc7e9b47a00b3156R884

`.addMarker()` is expecting an object with the properties `lat` and `lng`. We're passing an object with the property `_objectInstance` and two getter functions that will bring the values of `lat` and `lng`. Maybe the `.addMarker()` function does not work with getters and it needs static properties. 

If it works with 2.0.3 but not the latest version, then this is most likely the issue.

I will look into it soon.

**Comment by ihadeed** [comment]
Should be fixed now, please update to `ionic-native@2.1.8`

**Comment by ColinFrick** [comment]
Thanks @ihadeed It works now.
